luminoth/tools/dataset/readers/object_detection/imagenet.py

```diff
@@ -0,0 +1,209 @@
+                        image_id))
+                self.errors += 1
+                continue
+
+            objects = annotation.get('object')
+            if objects is None:
+                # If there's no bounding boxes, we don't want it.
+                continue
+
+            # TODO: consider alternatives to using Pillow here.
+            image_pil = Image.open(image_path)
+            width = image_pil.width
+            height = image_pil.height
+            image_pil.close()
+
+            gt_boxes = []
+            for b in annotation['object']:
+                try:
+                    label_id = self.classes.index(self._wnids[b['name']])
+                except ValueError:
+                    continue
+
+                (xmin, ymin, xmax, ymax) = self._adjust_bbox(
+                    xmin=int(b['bndbox']['xmin']),
+                    ymin=int(b['bndbox']['ymin']),
+                    xmax=int(b['bndbox']['xmax']),
+                    ymax=int(b['bndbox']['ymax']),
+                    old_width=int(annotation['size']['width']),
+                    old_height=int(annotation['size']['height']),
+                    new_width=width, new_height=height
+                )
+
+                gt_boxes.append({
+                    'label': label_id,
+                    'xmin': xmin,
+                    'ymin': ymin,
+                    'xmax': xmax,
+                    'ymax': ymax,
+                })
+
+            if len(gt_boxes) == 0:
+                continue
+
+            self.yielded_records += 1
+
+            yield {
+                'width': width,
+                'height': height,
+                'depth': 3,
+                'filename': annotation['filename'],
+                'image_raw': image,
+                'gt_boxes': gt_boxes,
+            }
+
+    def _is_valid(self):
+        if not tf.gfile.Exists(self._data_dir):
+            raise InvalidDataDirectory(
+                '"{}" does not exist.'.format(self._data_dir)
+            )
+
+        if not tf.gfile.Exists(self._imagesets_path):
+            raise InvalidDataDirectory('ImageSets path is missing')
+
+        if not tf.gfile.Exists(self._images_path):
+            raise InvalidDataDirectory('Images path is missing')
+
+        if not tf.gfile.Exists(self._annotations_path):
+            raise InvalidDataDirectory('Annotations path is missing')
+
+    def _get_split_path(self):
+        return os.path.join(
+            self._imagesets_path, '{}.txt'.format(self._split)
+        )
+
+    def _get_image_path(self, image_id):
+        return os.path.join(
+            self._images_path, '{}.JPEG'.format(image_id)
+        )
+
+    def _get_image_annotation(self, image_id):
+        return os.path.join(self._annotations_path, '{}.xml'.format(image_id))
+
+    def _get_record_names(self):
+        split_path = self._get_split_path()
+
+        if not tf.gfile.Exists(split_path):
+            raise ValueError('"{}" not found'.format(self._split))
```

Comment:
Not sure if `ValueError` is best exception here.

Reply:
Alternative?